From 15ea34224e7e3ebd6025aab4a01e963095c350c0 Mon Sep 17 00:00:00 2001
From: Carl Shaw <car.shaw@st.com>
Date: Wed, 3 Jun 2009 14:06:07 +0100
Subject: [PATCH] stm_i2c: Add write barrier to pio i2c driver

This patch does a read of the PIO state after each write to force
synchronisation.  As writes are posted, we cannot otherwise
guarantee that a write has actually occurred.

Signed-off-by: Peter Bennett <peter.bennett@st.com>
---
 drivers/i2c/busses/i2c-st40-pio.c |    2 ++
 1 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/drivers/i2c/busses/i2c-st40-pio.c b/drivers/i2c/busses/i2c-st40-pio.c
index 5ed6a05..98c1574 100644
--- a/drivers/i2c/busses/i2c-st40-pio.c
+++ b/drivers/i2c/busses/i2c-st40-pio.c
@@ -43,6 +43,7 @@ static void bit_st40_pio_setscl(void *data, int state)
 	struct ssc_pio_t *pio_info =
 		(struct ssc_pio_t *)pdev->dev.platform_data;
 	stpio_set_pin(pio_info->clk, state);
+	stpio_get_pin(pio_info->clk);
 }
 
 static void bit_st40_pio_setsda(void *data, int state)
@@ -51,6 +52,7 @@ static void bit_st40_pio_setsda(void *data, int state)
 	struct ssc_pio_t *pio_info =
 		(struct ssc_pio_t *)pdev->dev.platform_data;
 	stpio_set_pin(pio_info->sdout, state);
+	stpio_get_pin(pio_info->sdout);
 }
 
 static int bit_st40_pio_getscl(void *data)
-- 
1.6.0.6

